plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPartitioningHandle.java

```diff
@@ -64,26 +66,45 @@ private static Map<Integer, List<Integer>> buildDataPaths(PartitionSpec spec)
     {
         Set<Integer> partitionFieldIds = spec.fields().stream().map(PartitionField::sourceId).collect(toImmutableSet());
 
-        int channel = 0;
         Map<Integer, List<Integer>> fieldInfo = new HashMap<>();
         for (Types.NestedField field : spec.schema().asStruct().fields()) {
```

Comment:
After reading the change again, it seems that the sole purpose of the change is to ensure that channels are allocated in field ID order.  If so, can we fix this by sorting the fields before this loop.  I'm thinking something like this:

```suggestion
        List<Types.NestedField> fields = spec.schema().asStruct().fields().stream()
                .sorted(Comparator.comparingInt(Types.NestedField::fieldId))
                .collect(toImmutableList());
        for (Types.NestedField field : fields) {
```

Reply:
it needs to be sorted by the field that used for partitioning, not only using the root field. 
for nested schema with [f1: [f3, f4], f2], and partition by f3 and f2; then order should be [f2, f1.f3]

I think it's sorted [here](https://github.com/trinodb/trino/blob/cb7a9cc1cf4e9ea26ccaa966e8a421a76c273954/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1241)

Reply:
Can you add a test that partitions on `[f4, f2, f3]`? I'm concerned about what happens when there are multiple nested partitioned fields, intermixed with non-nested or nested from another structured field.

Reply:
Added test with
```
WITH (partitioning = ARRAY[
                    '"r1.f2"',
                    'bucket(b1, 4)',
                    '"r1.f1"'
                ]) AS
                SELECT
                    CAST(ROW('f1', 'f2') AS ROW(f1 VARCHAR, f2 VARCHAR)) as r1
                    , CAST('b' AS VARCHAR) as b1
```